nbexchange/plugin/collect.py

```diff
@@ -30,17 +30,31 @@ def download(self, submission, dest_path):
         self.log.debug(
             f"Got back {r.status_code}  {r.headers['content-type']} after file download"
         )
-        tgz = r.content
 
-        try:
-            tar_file = io.BytesIO(tgz)
-            with tarfile.open(fileobj=tar_file) as handle:
-                handle.extractall(path=dest_path)
-        except Exception as e:  # TODO: exception handling
-            if hasattr(e, "message"):
-                self.fail(e.message)
+        if r.status_code > 399:
+            self.fail(
+                f"Error looking for assignments to collect: status code {r.status_code}"
```

Comment:
I think its worth adding some more data here like the course code and user id so we can investigate problems